### ISSUE TITLE
Make Install error

### ISSUE THREAD

**Issue by NVG2021** [lwfinger/rtl8192du#89]
Dear developer, I've tried to install the driver for the wlan that is installed on my laptop and I get this message, after  the initial download
aduser@prixton:~$ git clone https://github.com/lwfinger/rtl8192du.git
Cloning into 'rtl8192du'...
remote: Enumerating objects: 7709, done.
remote: Counting objects: 100% (56/56), done.
remote: Compressing objects: 100% (42/42), done.
remote: Total 7709 (delta 24), reused 30 (delta 14), pack-reused 7653
Receiving objects: 100% (7709/7709), 13.29 MiB | 4.45 MiB/s, done.
Resolving deltas: 100% (6151/6151), done.
aduser@prixton:~$ cd rtl8192du/
aduser@prixton:~/rtl8192du$ sudo make install
[sudo] password for aduser: 
install -p -m 644 8192du.ko  /lib/modules/5.13.0-28-generic/kernel/drivers/net/wireless/
install: cannot stat '8192du.ko': **No such file or directory
make: *** [Makefile:615: install] Error 1**

Thank you

**Comment by NVG2021** [comment]
sorry, my system is an UBUNTU based Laptop and intel celeron

**Comment by lwfinger** [comment]
You need to 'make' before 'sudo make install'

**Comment by NVG2021** [comment]
yes, i saw it... sorry
but now I've got another one
**Skipping BTF generation for /home/aduser/rtl8192du/8192du.ko due to unavailability of vmlinux**
make[1]: Leaving directory '/usr/src/linux-headers-5.13.0-28-generic'
aduser@prixton:~/rtl8192du$ sudo make install
[sudo] password for aduser: 
install -p -m 644 8192du.ko  /lib/modules/5.13.0-28-generic/kernel/drivers/net/wireless/
/sbin/depmod -a 5.13.0-28-generic
d**epmod: ERROR: failed to load symbols from /lib/modules/5.13.0-28-generic/kernel/drivers/input/misc/drv2667.ko: Invalid argument**
aduser@prixton:~/rtl8192du$

**Comment by lwfinger** [comment]
You need to look at the output of dmesg to see the reason for the 'invalid argument'. You can ignore the BTF generation problem. That is just a warning.

**Comment by NVG2021** [comment]
Hello, I guess that for me this is getting too hard to follow.... or understand. 
I can follow the copy/paste (and with my own errors) but what you suggest delivers lots of lines which for me could be chinese or arabic.....
I've copied the dmesg output to a file and maybe it can be understood with your eyes?
thank you so much
[mydmesg.log](https://github.com/lwfinger/rtl8192du/files/7998697/mydmesg.log)

**Comment by lwfinger** [comment]
Your dmesg output shows the BT part of the device, but nothing concerning the wifi.

Actually, that error is due to driver drv2667.ko, not this one. I think you need to reinstall the kernel, and likely the kernel firmware. I do not use Ubuntu, thus I cannot tell you what to do, but https://serverok.in/reinstall-kernel-in-ubuntu seems to have instructions, at least for the kernel.

One other thing of concern is the following section in your dmesg output:
[    1.733726] [Hardware Error]: event severity: fatal
[    1.733728] [Hardware Error]:  Error 0, type: fatal
[    1.733730] [Hardware Error]:   section_type: Firmware Error Record Reference
[    1.733731] [Hardware Error]:   Firmware Error Record Type: SOC Firmware Error Record Type1 (Legacy CrashLog Support)
[    1.733732] [Hardware Error]:   Revision: 0
[    1.733733] [Hardware Error]:   Record Identifier: 3000100000000

I have never seen this kind of hardware error in any of my logs. If this laptop is new and can be returned, then I suggest that. Otherwise, you could find a real computer guru that might understand these errors better than I do.

**Comment by NVG2021** [comment]
is this driver set for fedora? or which distro is it for.... as I can re-install easily and have no data yet
Thanks for the input, which is far more than I can get myself
this laptop is very weird and can't be returned, being almost new....
I may change the distro so I can give it a last try....  Many thanks for your time and valuable knowledge

**Comment by NVG2021** [comment]
and I guess I did not proper install... on the README there is a step which is titled 
For Fedora..... and I did not perform that step is it also needed for Ubuntu?
thanks

**Comment by lwfinger** [comment]
The driver is agnostic as regards the distro. Those sections of the README are there to tell the user how to install the various packages needed to build the driver. The driver actually depends on the kernel, which is essentially the same for all distros.
